### PR TITLE
fix: proper last \r evaluation

### PR DESCRIPTION
**Pull request by gescheit**
fix: proper last \r evaluation

### CODE REVIEW

File: pkg/terminal/parser.go [hunk 1]
```diff
@@ -136,10 +147,21 @@ func (m *Parser) parse() ([]byte, error) {
 				m.pos = begin - 1
 
 			case ELINE, SGR:
-				m.data = sliceEdit(m.data, escStart, m.pos+1)
+				// Erases part of the line.
+				// If n is 0 (or missing), clear from cursor to the end of the line.
```
Comment (raydent) [inline pkg/terminal/parser.go:151]: What is an 'n'?

File: pkg/terminal/parser.go [hunk 2]
```diff
@@ -136,10 +147,21 @@ func (m *Parser) parse() ([]byte, error) {
...
+				// If n is 1, clear from cursor to beginning of the line.
+				// If n is 2, clear entire line. Cursor position does not change.
+				// only n == 0 is supported TODO: check parameter var
+				m.data = sliceEdit(m.data, escStart, m.pos+1) // drop esc
 				// Rolling back to the place of the deleted esc-1, consume moved to the esc old pos
 				// Necessary in case of two esc sequences in a row
-				m.pos = escStart - 1
+				m.pos = escStart
+				//lineEnd := bytes.IndexByte(m.data[0:m.pos], NEWLINE)
```
Comment (raydent) [inline pkg/terminal/parser.go:159]: Is it still needed?

File: pkg/terminal/parser.go [hunk 3]
```diff
@@ -44,6 +44,17 @@ func Parse(data []byte) ([]byte, error) {
 	return parser.parse()
 }
 
+// ParseDropLastReturn evaluates ANSI escape sequences and apply logik related to prompt cut
```
Comment (raydent) [inline pkg/terminal/parser.go:47]: logic*

File: pkg/terminal/parser.go [hunk 4]
```diff
@@ -187,3 +209,12 @@ func min(a, b int) int {
 	}
 	return b
 }
+
+func dropLastR(input []byte) []byte {
```
Comment (raydent) [inline pkg/terminal/parser.go:213]: Maybe trimAfterLastNewline is a better name?

**Review by raydent — CHANGES_REQUESTED** [review]

File: pkg/terminal/parser.go [hunk 5]
```diff
@@ -136,10 +147,21 @@ func (m *Parser) parse() ([]byte, error) {
...
 
 			case ELINE, SGR:
-				m.data = sliceEdit(m.data, escStart, m.pos+1)
+				// Erases part of the line.
+				// If n is 0 (or missing), clear from cursor to the end of the line.
+				// If n is 1, clear from cursor to beginning of the line.
+				// If n is 2, clear entire line. Cursor position does not change.
+				// only n == 0 is supported TODO: check parameter var
+				m.data = sliceEdit(m.data, escStart, m.pos+1) // drop esc
 				// Rolling back to the place of the deleted esc-1, consume moved to the esc old pos
```
Comment (raydent) [inline pkg/terminal/parser.go:156]: Is it no longer true? On line 158 we roll back to esc

**Review by raydent — APPROVED** [review]

File: pkg/terminal/parser.go (hunk 1 above)
Comment (gescheit) [reply pkg/terminal/parser.go:151]: "n" stands for number value. Full specification is "CSI n K"

**Pull request merged**